test.sh

```diff
@@ -1,5 +1,67 @@
 #!/bin/bash
 
 # main test runner for heketi
+# Executes all executable scripts under tests dir
+# in sorted order.
 
-# TODO: add tests here...
+vecho () {
+	if [ "$verbose" = "yes" ] ; then
+		echo "$*"
+	fi
+}
+
+CLI="$(getopt -o xvh --long exitfirst,verbose,help -n $0 -- "$@")"
+eval set -- "$CLI"
+while true ; do
+	case "$1" in
+		-x|--exitfirst)
+			exitfirst=yes
+			shift
+		;;
+		-v|--verbose)
+			verbose=yes
+			shift
+		;;
+		-h|--help)
+			echo "$0 [options]"
+			echo "  Options:"
+			echo "    -x|--exitfirst    Exit on first test failure"
+			echo "    -h|--help         Display help"
+			exit 0
+		;;
+		--)
+			shift
+			break
+		;;
+		*)
+			echo "unknown option" >&2
+			exit 2
+		;;
+	esac
+done
+
+SCRIPT_DIR="$(cd "$(dirname "${0}")" && pwd)"
+
+# environment vars exported for test scripts
+# (this way test scripts dont need cli parsing, we do it here)
+export HEKETI_TEST_EXITFIRST=$exitfirst
+export HEKETI_TEST_SCRIPT_DIR="$SCRIPT_DIR"
+
+cd "$SCRIPT_DIR"
+failed=0
+for tname in $(ls tests | sort) ; do
+	tpath="./tests/$tname"
+	if [ -f $tpath -a -x $tpath ]; then
+		vecho "-- Running: $tname --"
```

Comment:
The hunk in the loop (inside the if block) could nicely go into a function (run_test or similar)

Reply:
OK, PR  #896 went with "function run-test()" do you want to use that convention here or do you prefer underscore? (I am somewhat more familiar naming shell functions with underscores but would be happy either way)

Reply:
I prefer underscores. looks more like real programming ;-)

Reply:
and i prefer a multi-line function over a 1-line construct for readability :-)

Reply:
we can also do the refactoring in a follow up commit if you feel exhausted. ;-)
But I would appreciate if the request for variable and conditials be folllowed:

> Generally as a style, please try using curly braces for variable expansion (like ${foobar} instead of $foobar), and double brackets for if: if [[ -n ${foobar} ]]; ...

Reply:
> we can also do the refactoring in a follow up commit if you feel exhausted. ;-)

Nah, not yet. :-)
Now that I've got your feedback I'll update the series.